### PR TITLE
Create member.ts

### PR DESCRIPTION
**Pull request by KevinBacabac**
## Summary of Changes
Add a .member command which takes in a UW ID and outputs whether it is a CSC member.

## Related Issues
Resolves https://github.com/uwcsc/codeybot/issues/241

## Steps to Reproduce
Try .member with no arguments, a valid user id and an invalid user id.

## Demonstration of Changes
image

### CODE REVIEW

File: src/commands/miscellaneous/member.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { Command, container } from '@sapphire/framework';
```
Comment (mcpenguin) [inline src/commands/miscellaneous/member.ts:1]: Put the details in the `commandDetails` folder and the codey command in the `commands` folder to make it consistent with the rest of the code.

**Review by mcpenguin — CHANGES_REQUESTED** [review]
small changes

File: src/commands/miscellaneous/member.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+const MEMBER_API = 'https://csclub.uwaterloo.ca/~a3thakra/csc/adi-member-json-api/api/members.json';
+
+interface memberStatus {
+  name: string;
+  id: string;
+  program: string;
+}
+
+type UWIdType = string | undefined;
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts]: Could this be `UwIdType` instead?

File: src/commands/miscellaneous/member.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+/*
+ * Get member embed
+ */
+export const getMemberEmbed = async (uwid: UWIdType): Promise<MessageEmbed> => {
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts]: Could this be `uwId` instead?

File: src/commands/miscellaneous/member.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  if (foundMember) {
+    return new MessageEmbed()
+      .setColor('GREEN')
+      .setTitle(title)
+      .setDescription(`You're a CSC member! Hooray! ${getEmojiByName('codeyLove')}`);
+  }
+
+  const NOT_MEMBER_DESCRIPTION = `You're not a CSC member! ${getEmojiByName('codeySad')}
+
+Being a CSC member comes with gaining access to CSC machines, cloud, email, web hosting, and more! Additional details can be found here https://csclub.uwaterloo.ca/resources/services/
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts]: Could there be a period or exclamation mark after `found here`?

File: src/commands/miscellaneous/member.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  return new MessageEmbed().setColor('RED').setTitle(title).setDescription(NOT_MEMBER_DESCRIPTION);
+};
+
+const executeCommand: SapphireMessageExecuteType = async (
+  _client,
+  messageFromUser,
+  args,
+  _initialMessageFromBot
+): Promise<SapphireMessageResponse> => {
+  let uwid: UWIdType;
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:21]: Same thing here in terms of capitalization.

File: src/commands/miscellaneous/member.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    const messageArgs = content.split(' ').filter((m) => m != '.member');
+    if (messageArgs.length == 1) uwid = messageArgs[0];
+  } else if ('uwid' in args) {
+    uwid = args['uwid'] as string;
+  }
+  const memberEmbed = await getMemberEmbed(uwid);
+  return { embeds: [memberEmbed] };
+};
+
+const helpCommandDetails: CodeyCommandDetails = {
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:33]: This should be `memberCommandDetails`?

File: src/commands/miscellaneous/member.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  name: 'member',
+  aliases: [],
+  description: 'Get CSC Member information',
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:36]: Could this be `Gets CSC membership information` instead?

File: src/commands/miscellaneous/member.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  detailedDescription: `**Examples:**
+\`${container.botPrefix}member [id]\``,
+
+  isCommandResponseEphemeral: false,
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:40]: Could this be `true` instead?

File: src/commands/miscellaneous/member.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  messageWhenExecutingCommand: 'Getting Member Info...',
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:41]: Could this be `Getting CSC membership information...` instead?

File: src/commands/miscellaneous/member.ts [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  executeCommand: executeCommand,
+  messageIfFailure: 'Could not retrieve URL to the wiki page.',
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:43]: Could this be `Could not retrieve CSC membership information.` instead?

File: src/commands/miscellaneous/member.ts [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  codeyCommandResponseType: CodeyCommandResponseType.EMBED,
+
+  options: [
+    {
+      name: 'uwid',
+      description: 'A UW alphanumerical ID',
```
Comment (Picowchew) [inline src/commands/miscellaneous/member.ts:49]: Could this be `Quest ID` instead?

**Review by Picowchew — CHANGES_REQUESTED** [review]
I just left various super small comments.

**Review by Picowchew — CHANGES_REQUESTED** [review]
To be more clear, `memberCommandDetails` and `executeCommand` should go into the `commandDetails` folder. You can follow the `coin` commands as an example for where things should go. Commands without subcommands, such as `help`, currently have their `commandDetails` in the `commands` folder, but we are planning on moving them to the `commandDetails` folder sometime in the near future for consistency.

Also, it looks like there are some linter warnings.

**Review by Picowchew — APPROVED** [review]
LGTM!

**Review by mcpenguin — APPROVED** [review]
LGTM too!

File: src/commandDetails/miscellaneous/member.ts [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+import {
+  CodeyCommandDetails,
+  CodeyCommandOptionType,
+  CodeyCommandResponseType,
+  SapphireMessageExecuteType,
+  SapphireMessageResponse
+} from '../../codeyCommand';
+import { getEmojiByName } from '../../components/emojis';
+
+const MEMBER_API = 'https://csclub.uwaterloo.ca/~a3thakra/csc/adi-member-json-api/api/members.json';
```
Comment (neilparikh) [inline src/commandDetails/miscellaneous/member.ts:13]: This should be `https://csclub.uwaterloo.ca/api/members.json`

**Review by neilparikh — CHANGES_REQUESTED** [review]

**Review by neilparikh — APPROVED** [review]

**Pull request merged**